### PR TITLE
Network layer

### PR DESCRIPTION
**Pull request by saurabhalp**
added network helper, data repository, view model and displayed it using compose in screen.kt

### CODE REVIEW

**Review by sksinha2410 — APPROVED** [review]

File: .idea/.name [hunk 1]
```diff
@@ -0,0 +1 @@
+PGPod
```
Comment (Jha2508) [inline .idea/.name:1]: delete

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
```
Comment (Jha2508) [inline .idea/codeStyles/Project.xml:2]: delete

File: .idea/codeStyles/codeStyleConfig.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <state>
+    <option name="USE_PER_PROJECT_SETTINGS" value="true" />
```
Comment (Jha2508) [inline .idea/codeStyles/codeStyleConfig.xml:3]: delete

File: app/src/main/java/com/starstruckstech/pgpodandroid/DataRepository.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.starstruckstech.pgpodandroid
+
+class PostRepository(private val networkHelper: NetworkHelper) {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/DataRepository.kt:3]: either move it to `TestUtilities` or remove it

File: app/src/main/java/com/starstruckstech/pgpodandroid/PostViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.starstruckstech.pgpodandroid
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+
+class PostViewModel(private val repository: PostRepository) : ViewModel() {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/PostViewModel.kt:9]: same as above

File: app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.starstruckstech.pgpodandroid
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RetrofitClient {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt:6]: java docs

File: app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    private const val BASE_URL = "https://jsonplaceholder.typicode.com/"
+
+    val retrofitService: ApiServicee by lazy {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt:9]: comments for the usage

File: app/src/main/java/com/starstruckstech/pgpodandroid/apiService.kt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.starstruckstech.pgpodandroid
+
+import retrofit2.http.GET
+interface ApiServicee {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/apiService.kt:4]: spelling Mistake + java docs

File: app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        retrofit.create(ApiServicee::class.java)
+    }
+
+    private val retrofit by lazy {
+        Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+    }
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/RetrofitClient.kt:18]: Add the logic for public and private api calls according to the endpoints given

File: app/src/main/java/com/starstruckstech/pgpodandroid/NetworkHelper.kt [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package com.starstruckstech.pgpodandroid
+
+class NetworkHelper(private val apiService: ApiServicee) {
+    suspend fun getPosts(): List<Post> {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/NetworkHelper.kt:4]: use retrofit client and do the API Call and return the corresponding data class object for  a sample API

**Pull request merged**